lib/policies/key-auth/keyauth.js

```diff
@@ -12,24 +12,19 @@ passport.use(new LocalAPIKeyStrategy({ passReqToCallback: true }, (req, apikey,
 
   const keyParts = apikey.split(':');
 
-  // as per http://passportjs.org/docs/authorize
-  // after authentication  req.user is populated
-  // and after authorization req.account is filled in
   authService.authenticateCredential(keyParts[0], keyParts[1], credentialType)
     .then(consumer => {
       if (!consumer) {
         return done(null, false);
       }
       const endpointScopes = req.egContext.apiEndpoint.scopes && req.egContext.apiEndpoint.scopes.map(s => s.scope || s);
-      req.user = consumer;
 
       return authService.authorizeCredential(keyParts[0], credentialType, endpointScopes)
         .then(authorized => {
           if (!authorized) {
-            return done(null, false);
+            return done(null, false, {unauthorized: true});
```

Comment:
Note: check who's using this parameter. Shouldn't `false` be enough?

Reply:
Third parameter `info`. It is used in common auth handler. 
it is a way to understand is it 401 or 403. 
Completely custom logic, if you have better idea we can discuss. 

info is a helper kind of a  `state`  parameter is passport. Typically used for messages. 

check out how all three strategies use ActionParams common logic.